app/main.py

```diff
@@ -1,3 +1,24 @@
 def cache(func):
-    # Write your code here
-    pass
+    cached_data = {}
+
+    def wrap(*args):
+        nonlocal cached_data
+        for key, value in {args: func(*args)}.items():
+            if str(key) not in cached_data:
+                cached_data.update({f"{args}": func(*args)})
```

Comment:
```suggestion
                cached_data.update(args: func(*args)})
```
Converting to string isn't correct because you would consider a tuple (1, 2) and a string "(1, 2)" as the same args